### DISCUSSION TITLE
Changing the program, a dynamic GUI... a discussion only

### DISCUSSION THREAD

**Discussion by scotthibbs** [w7sst/MorseRunner#130] [Ideas]
We got a couple discussions on changing the program. For example adding Farnsworth or adding a pause button. We already agreed on adding a better CW speed control.

As for now, we just want to concentrate on adding contests. This discussion is way ahead of any action at this time but it is very important we have this discussion.

I was watching this video by HB9DHG Fulvio he posted 6 days ago doing a HST contest in 1.68 (win10) with activity of 4 for 10 minutes. https://youtu.be/IlP7a_-ooEA Notice his use of the page up and page down buttons. This changes the CW Speed control. He can read calls in a pile up at 70 wpm and he reads numbers at 45 wpm but sends code at 120 wpm (the highest setting). His score is 3604 and achieves a 372qso/hour rate.

Obviously, this is not realistic since it is unreasonable that all 60ish contacts knew to send him 70wpm calls, with 45wpm exchanges, all the while copying him at 120 wpm perfectly and amazingly have no complaint about it. Because if we were realistic, Fulvio would be the LID for sending faster 120wpm code (no offense Fulvio). But provided the functionality of the 1.68 program this is altogether real and so is his score a real one. We just watched it. Fulvio is absolutely an HST Champ on 1.68 and this talent on MR is extremely impressive.

We can be confused when it comes to realism and to which it refers, real contest simulation and real as in original to 1.68. We need to keep this in mind for our discussion.

Our readme welcome says MR is "...an effective practice tool for improving their CW copying and pileup handling skills." and our first goal is to "Improve end-user experience by supporting multiple popular contests." I think we all want to improving CW skills and the end-user experience (aka fun) with MR. We have to change the GUI and the backbone of 1.68 to add more contests. This is a given. 

So the question for discussion is how much do we want to keep to the original 1.68 because if we make it more contest real then Fulvio will not want to use our new program. And I think we all want the community as a whole to use our new and improved version. We want to provide more but stay true to the original program. 

I'm thinking we can do this and preserve the 1.68 experience where we can. Since we are adding real contests, maybe we need to create a real contest mode (e.g. Real Field Day vs Practice Field Day modes). In a real contesting mode the speed control, Farnsworth, pause and other fun (aka end-user experience) add-ons are disabled like in "real" life. In practice modes we can provide a ton of fun. If we correct the HST mode to "make it more real" by disabling the speed control then we need to have a separate mode for "HST original 1.68" so Fulvio can also have fun. And we can attempt to beat him using our Community Edition.  

Change is hard. This program is world wide and the larger community uses the original version. Thus if we keep the original modes of the original program (HST Competition original, WPX Competition Original) while providing fun (additional contests in a real mode, or a practice mode with a lot of add-ons) this will allow the user base to change with us, experiment and increase playability. 

Just an idea. Yours are welcome and I'd love to hear them.

**Comment by K6OK** [comment]
Thoughtful post; it's good to reflect on where MRCE should be going.  My thoughts:

- MRCE should focus on serving the largest user base.  In about 10 days CQ WW CW contest will be on the air with roughly 9,000 stations across the globe.  The HST community is tiny by comparison.  So if we want to maximize MRCE adoption then contest simulation should be the primary focus.  That's not to say HST should be eliminated or ignored, only that it should be a lower priority.  W7SST's Roadmap as posted has it right.

- I don't believe 1.68 functionality should be a goal.  If there are GUI changes that improve ergonomics then MRCE should adopt them.  I've been using MRCE daily to prepare for CQ WW and I'll share some thoughts later on possible GUI improvements.

- The IARU HST championships use 1.68 currently.  They aren't going to switch to MRCE if the CE edition merely adopts the same functions and features.  If we want IARU HST to adopt MRCE as the "official" competition software, then what I'd propose is to reach out to the IARU organizers and ask them what features or changes they want for their in-person competitions.

- MRCE should not cater to edge use cases that 99%+ of the user base won't use.   If those edge users want to stick with 1.68, that's fine, they can do that.  

Just my two cents, others may have different ideas.  73 de Yagi-Berra

**Comment by tekenny** [comment]
I do think if you want to be listed on the scoreboard it needs to be a consistent (static during the run) set of settings. Although MR can be used in some ways to improve CW copy that isn't it's primary mission which actually is improving CW contesting skills. So I don't think we have to be so strict that we remove the ability that Fulvio is using but that sort of run shouldn't be on a scoreboard.

**Comment by scotthibbs** [comment]
Thanks @Yagi-Berra for the input. When you stated earlier: "The IARU HST championships use 1.68 currently. They aren't going to switch to MRCE if the CE edition merely adopts the same functions and features. If we want IARU HST to adopt MRCE as the "official" competition software, then what I'd propose is to reach out to the IARU organizers and ask them what features or changes they want for their in-person competitions."

Having no knowledge of this, I've done some research. It seems the IARU HST Championship still uses 1.67. I'm not exactly sure why they didn't use Alex's 1.68, maybe it was to keep all the previous records on the same playing field? (I can't remember what the difference is between 1.67 and 1.68 currently but don't remember it being a big change.) It could be if they switch to CE, they might want the same functions and features after all. They do use a new list file for each competition, which makes sense. 

So HB9DHG Fulvio is as real as it gets - at least according to the rules. (I did assume he was using 1.68, most likely 1.67.) So it's on a scoreboard. A big scoreboard because MR (with others) is used to determine the High Speed Telegraphy World Champion and possible World Records in copying code. 

The most interesting thing here is that this program's "HST competition" mode is the competition, not a real world radio contest. I find this ironic and amusing because it is like me. (I've been learning Morse Code longer than I've been a ham (since '92) and later passed Tech Plus (5wpm) just for 10M HF voice privileges.) I don't contest, nor do I use CW on the air. I suggested a Field Day Contest so I could sit with other CW operators during FD and stab at the key with lots of hand holding, Elmering and emotional support, or maybe even short "thanks for trying, now get out the way." :) I use this program simply to learn code better, faster, and fun-er-er. I'm sure I'm not alone.

In closing here, I wanted to share their settings  and rules according to the rule book if you are interested: 

I. SETTINGS for ”Morse runner v1.67”
The program shall be used with HST competition file
The main menu shall be set as follows:
[System]
Buf Size=3
[Station]
Call=optional
Name=optional
Pitch=optional
Band Width=10
Wpm=optional
QSK=0
Rules for High Speed Telegraphy Championships 17
Self Mon Volume= 0
Save Wav=0
[Band]
Activity=4
QRN=0
QRM=0
QSB=0
Flutter=0
Lids=0
[Contest]
Duration=10
Hi Score=0
Competition Duration=10
All features must be set and checked in configuration file by the jury before the competition. " 

The competitor is allowed to complete two attempts. Each attempt is limited to 10 minutes. The competitor has 25 minutes to stay in the test room. After the end of this period the attempt will be stopped, and the better score shall be taken as the competitor's entry. All competitors from one category will compete at the same time. If a competitor is making deliberate noise to the other competitor, he will be disqualified and no points will be awarded for that test. The keyboard must be fit to the US standard (QWERTY). It is NOT allowed to use own headphones and keyboard. Competitors will compete under their own names or call signs. Contestants have the right to prepare for his/her attempt, i.e. to make all possible settings and a test run.
Time for these preparations is limited up to three minutes starting with entering the room. Before he/she starts the attempt, he/she should sign the document that everything is working properly. If the referee states that preparation-/break-time is over,
contestant must start attempt immediately.

**Reply by K6OK** [reply]
Thank you for the detailed HST rules post.  Perhaps the best thing, down the road after new contests have been added, would be a special fork just for HST competition purposes.  The HST version would have no contests, just HST, and it would lock all the settings that the HST administrators want.  This would be done in cooperation with the IARU HST people, of course.

Under this proposal, MRCE would still support "casual" HST as it does now.  There would also be a new program Morse Runner HST Competition Edition,  endorsed and used by IARU HST.

**Comment by tekenny** [comment]
Rather than a repo fork I'd like us to use the same repo base but separate build scripts for the MR vs HST versions. I think a fork or even branch could make keeping the two resulting programs more complex. So same code base but separate executable builds would be the way to manage it the general contesting need as well as the HST needs.

**Reply by K6OK** [reply]
@tekenny, very good idea, I concur.
